### PR TITLE
add a new file for fortios_configuration_fact and edit the generate_modules.py file

### PR DESCRIPTION
**Pull request by JieX19**
A few examples to test the module.

```
- hosts: fortigateslab
  connection: httpapi
  collections:
    - fortinet.fortios
  vars:
    ansible_httpapi_use_ssl: yes
    ansible_httpapi_validate_certs: no
    ansible_httpapi_port: 443
    vdom: "root"
  tasks:
  - name: Get a list
    fortios_configuration_fact:
      vdom: "{{  vdom }}"
      access_token: "{{ fortios_access_token }}"
      selector: firewall_address
      # Will get a list of results if mkey is not provided
  - name: Get a list
    fortios_configuration_fact:
      vdom: "{{  vdom }}"
      access_token: "{{ fortios_access_token }}"
      selector: firewall_address
      # Will get a list of results if params is an empty object
      params: {}
  - name: single get
    fortios_configuration_fact:
      vdom: "{{  vdom }}"
      access_token: "{{ fortios_access_token }}"
      selector: firewall_address
      # optionally list or single get
      params:
        name: "gmail.com"
  - name: single get
    fortios_configuration_fact:
      vdom: "{{  vdom }}"
      access_token: "{{ fortios_access_token }}"
      selector: firewall_address
      # unmatched type
      params:
        name: 12
  - name: single get
    fortios_configuration_fact:
      vdom: "{{  vdom }}"
      access_token: "{{ fortios_access_token }}"
      selector: firewall_address
      # nonexistent name
      params:
        name: "gmail.co"
```

### CODE REVIEW

File: ansible_templates/fact.j2 [hunk 1]
```diff
@@ -0,0 +1,149 @@
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.connection import Connection
+from ansible_collections.fortinet.fortios.plugins.module_utils.fortios.fortios import FortiOSHandler
+from ansible_collections.fortinet.fortios.plugins.module_utils.fortimanager.common import FAIL_SOCKET_MSG
```
Comment (chillancezen) [inline ansible_templates/fact.j2:5]: Add one line before(legacy fortiosapi has been deprecated)
```
from ansible_collections.fortinet.fortios.plugins.module_utils.fortios.fortios import check_legacy_fortiosapi
```
https://github.com/fortinet-ansible-dev/fortios-ansible-generator/blob/feature/1.x/ansible_templates/code.j2#L5

File: ansible_templates/fact.j2 [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+        fos = FortiOSAPI()
+
+        login(module.params, fos)
+        {% if supports_check_mode -%}
+        is_error, has_changed, result = fortios_configuration_fact(module.params, fos, module.check_mode)
+        {% else -%}
+        is_error, has_changed, result = fortios_configuration_fact(module.params, fos)
+        {% endif -%}
+        fos.logout()
```
Comment (chillancezen) [inline ansible_templates/fact.j2:132]: https://github.com/fortinet-ansible-dev/fortios-ansible-generator/blob/feature/1.x/ansible_templates/code.j2#L365

remove legacy fortiosapi code.

member to append another parameter when constructing `FortiOSHandler`
```
fos = FortiOSHandler(connection, module)
```

File: ansible_templates/fact.j2 [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    password = data['password']
+    ssl_verify = data['ssl_verify']
+
+    fos.debug('on')
+    if 'https' in data and not data['https']:
+        fos.https('off')
+    else:
+        fos.https('on')
+
+    fos.login(host, username, password, verify=ssl_verify)
```
Comment (chillancezen) [inline ansible_templates/fact.j2:29]: not used anymore in httpapi only mode.

**Review by chillancezen — COMMENTED** [review]
mostly deprecating legacy fortiosapi code.

File: ansible_templates/fact.j2 [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+            "required": True,
+            "type": "str",
+            "options": [
+                {% for selector in selector_definitions.keys() -%}
+                "{{selector}}",
+                {% endfor -%}
+            ],
+        }
+    }
+
```
[the remark this answers is not in the document]
Comment (JieX19) [reply ansible_templates/fact.j2:95]: Yes, I will. Thank you for your reminder!

File: ansible_templates/fact.j2 [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+    definition = MODULE_MKEY_DEFINITONS.get(selector, {})
+
+    if len(selector_params) == 0 or len(definition) == 0:
+        return True, {}
+
+    mkey = definition['mkey']
+    mkey_type = definition['mkey_type']
+    if mkey_type == str(None):
+        return False, {"message": "params are not allowed for " + selector}
+    mkey_value = selector_params.get(mkey)
```
Comment (chillancezen) [inline ansible_templates/fact.j2:47]:
```
mkey_value = selector_params.get(mkey_name) if selector_params else None
```
ditto.

**Review by chillancezen — COMMENTED** [review]
Support case where no `params` is given.

File: ansible_templates/fact.j2 (hunk 5 above)
Comment (chillancezen) [reply ansible_templates/fact.j2:47]: wrong place, moves to line 63

**Review by chillancezen — APPROVED** [review]
Verified. thanks JIe.

**Pull request merged**